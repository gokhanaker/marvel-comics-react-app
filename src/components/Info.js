import React from 'react';

// Stateless functional component as a presentational component
// ES6 Destructuring passed prop
const Info = ({superHeroName, superHeroDescription, superHeroImage} ) => {

    // If there is no such a superhero in the marvel api then display nothing
    // Conditional rendering
    if (!superHeroName || !superHeroDescription) 
        return null;

    return(
        <div id='info'>
            <h3>Marvel Superhero: {superHeroName} </h3>
            <p> {superHeroDescription} </p>
            <img src={superHeroImage} 
                style={{
                    width: 300,
                    height: 300
                 }}>
            </img>
        </div>
    )
}

export default Info;

