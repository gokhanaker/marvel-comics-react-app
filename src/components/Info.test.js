import React from 'react';
import renderer from 'react-test-renderer';
import Info from './Info';

describe('Testing Info Component', () => {
  
    // TODO This test is not correct re-implement it
    test('<Info />  it renders', () => {
        const wrapper = renderer.create(<Info  />);
        expect(wrapper.toJSON()).toMatchSnapshot();
    });
 
});