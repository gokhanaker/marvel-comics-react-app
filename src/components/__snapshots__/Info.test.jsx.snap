// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Info Component <Info />  it renders 1`] = `null`;
